### DISCUSSION TITLE
Load reusable components via renderComponent

### DISCUSSION THREAD

**Discussion by pplopx** [flyntwp/flynt#532] [Q&A] [answered]
I am using Flynt 2 and reusable components. I'm editing a custom page.twig template where I want to include a reusable component via renderComponent in all pages using that template.

But I don't know how to call it using renderComponent, I have tried with the component id (post id) value but I don't know the key. The component id should be able to add it directly in the template so I don't have to include it in all the posts that use that specific page template.

Could you please document a bit on how to insert static components via template? With the necessary key/value sets that can be set in the template itself or, in addition, via ACF custom meta fields.

Thanks!

**Comment by aaronmeder** [comment, marked as answer]
Hey @pplopx 

This should work inside your `page.twig`:

```twig
{{
    renderComponent(
      'ReusableComponent',
      {
        reusableComponent: get_post(10),
      }
    )
}}
```

Where `10` would be the ID of your reusable component post that you would like to render.

![Screenshot 2024-02-11 at 08 51 00](https://github.com/flyntwp/flynt/assets/5564163/9f06c408-f4a3-464e-9ee1-da4f3ac1c9de)

Is that what you were looking for?

**How to reconstruct**
To figure this out I've added a reusable component to my page (through the wp-admin editor) and used `{{ dump(component) }}` inside the pageComponents loop in page.twig to figure out the keys it would return.

All the best from 🇨🇭
Aaron

**Reply by pplopx** [reply]
Thanks @aaronmeder, It works great! I hadn't thought of using twig/dump, I was stuck looking for the same thing in WP metadata. Thank you for your time for such an enlightening explanation.

**Reply by aaronmeder** [reply]
@pplopx happy to hear it works! Would be great if you could mark the question as answered too.
